### PR TITLE
Added tests to increase coverage

### PR DESCRIPTION
**Pull request by SaidShah**
- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

Refactored some tests and added some more tests to improve code coverage.

### CODE REVIEW

File: src/utilities/scroll-utils.test.ts [hunk 1]
```diff
@@ -36,6 +36,35 @@ describe("ScrollUtils", () => {
...
+            const options = { initialDelay: 200 };
+            const elementId = faker.random.uuid();
+            const setTimeoutMock = jest.spyOn(window, "setTimeout");
+
+            // Act
+            ScrollUtils.scrollToElementById(elementId, options);
+
+            // Assert
+            expect(setTimeoutMock).toBeCalled();
+            expect(setTimeoutMock.mock.calls[0][1]).toBe(options.initialDelay);
```
Comment (HeyKos) [inline src/utilities/scroll-utils.test.ts:66]: I think you should consider using `expect(setTimeoutMock).toBeCalledWith (elementId, expect.objectContaining({initial delay: options.initialDelay}));`

File: src/utilities/scroll-utils.test.ts [hunk 2]
```diff
@@ -54,21 +83,26 @@ describe("ScrollUtils", () => {
...
-        });
-
-        it("when location is undefined, then returns", () => {
-            // Arrange & Act
-            const result = ScrollUtils.scrollToHash(undefined);
-
-            // Assert
-            expect(result).toBeUndefined();
-        });
+        test.each`
```
Comment (HeyKos) [inline src/utilities/scroll-utils.test.ts:86]: Nice use of `test.each` 😁

**Review by HeyKos — APPROVED** [review]
I'll approve, but I would like to see the one piece of feedback addressed if possible.

File: src/utilities/scroll-utils.test.ts (hunk 1 above)
Comment (brandongregoryscott) [reply src/utilities/scroll-utils.test.ts:66]: ☝️ I agree, using the `toBeCalledWith` matcher would be more intuitive

**Review by brandongregoryscott — CHANGES_REQUESTED** [review]
Let me know when that assertion is cleaned up 👍

**Comment by SaidShah** [comment]
@HeyKos @brandongregoryscott - The assertion is cleaned up if you would like to take another look. Thanks.

File: src/utilities/scroll-utils.test.ts [hunk 3]
```diff
@@ -36,6 +36,37 @@ describe("ScrollUtils", () => {
...
+            //Arrange
+            const options = { initialDelay: 200 };
+            const elementId = faker.random.uuid();
+            const setTimeoutMock = jest.spyOn(window, "setTimeout");
+
+            // Act
+            ScrollUtils.scrollToElementById(elementId, options);
+
+            // Assert
+            expect(setTimeoutMock).toBeCalledWith(
```
Comment (HeyKos) [inline src/utilities/scroll-utils.test.ts:65]: That's what I like to see! Nice job @SaidShah 🎉

**Review by HeyKos — APPROVED** [review]

File: src/utilities/scroll-utils.test.ts [hunk 4]
```diff
@@ -36,6 +36,37 @@ describe("ScrollUtils", () => {
             // Assert
             expect(getElementByIdMock).toBeCalledTimes(50);
         });
+
+        test("when element is not found after 50 attempts, it logs a console warning for development environment", () => {
```
Comment (wintondeshong) [inline src/utilities/scroll-utils.test.ts:40]: @SaidShah as an outsider reading this test (not knowing the implementation), I don't see where the '50 attempts' statement is tested? Is that something internal the implementation? If so, be aware that tests should try to be an "outsider" like I am. It shouldn't do things assuming internals of the tested party it has no control over. 

To reconcile the above, you might just simplify the test working by removing `after 50 attempts`?

File: src/utilities/scroll-utils.test.ts [hunk 5]
```diff
@@ -36,6 +36,37 @@ describe("ScrollUtils", () => {
...
+            jest.useFakeTimers();
+            ScrollUtils.scrollToElementById(elementId);
+            jest.runAllTimers();
+
+            // Assert
+            expect(consoleWarnMock).toBeCalled();
+        });
+
+        test("when scrollOption has initial delay, it calls setTimeout with supplied delay", () => {
+            //Arrange
```
Comment (wintondeshong) [inline src/utilities/scroll-utils.test.ts:56]: Nitpick: Space between `//` and `Arrange`

File: src/utilities/scroll-utils.test.ts [hunk 6]
```diff
@@ -36,6 +36,37 @@ describe("ScrollUtils", () => {
...
+            const options = { initialDelay: 200 };
```
Comment (wintondeshong) [inline src/utilities/scroll-utils.test.ts:57]: Suggestion: Consider using `faker.random.number(min: __, max: __)` (you can set the min and max).

**Review by wintondeshong — CHANGES_REQUESTED** [review]
@saidshahdev once the test wording around "50 attempts" is resolved, the rest are more food for thought and I'm good to approve

**Comment by SaidShah** [comment]
@wintondeshong I made the changes that you requested, thanks for catching those.

**Review by wintondeshong — APPROVED** [review]
@SaidShah thanks for making those changes!

**Review by brandongregoryscott — APPROVED** [review]

**Comment by brandongregoryscott** [comment]
@SaidShah Just a reminder to link the related issue in the description with [closing keywords](https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords) next time 👍 Thanks for the contribution!

**Pull request merged**